GlobalParams:
  NumCells: 10
  ObjThresh: 50
Pipeline:
  Pods:
    - Name: BlobEntryPoint
      Type: EntryPoint<Blob::Ptr>
      Params: ~
      Inputs: ~
    - Name: BlobProjector
      Type: BlobProjector
      Params: ~
      Inputs:
        Blob: BlobEntryPoint.Output
    - Name: GravitationalCloudOrienter
      Type: GravitationalCloudOrienter
      Params: ~
      Inputs:
        ProjectedBlob: BlobProjector.ProjectedBlob
    - Name: CloudOrienter
      Type: CloudOrienter
      Params:
        OrientCloud: False
      Inputs:
        ProjectedBlob: BlobProjector.ProjectedBlob
    - Name: OrientedBoundingBoxSize
      Type: BoundingBoxSize
      Params: ~
      Inputs:
        Cloud: GravitationalCloudOrienter.OrientedCloud
    - Name: OrientedNormalizedDensityHistogram
      Type: NormalizedDensityHistogram
      Params:
        NumBins: 10
      Inputs:
        Cloud: GravitationalCloudOrienter.OrientedCloud
    - Name: HSVHistogram
      Type: HSVHistogram
      Params:
        NumBins: 10
        ValueThreshold: 0.2
        SaturationThreshold: 0.2
      Inputs:
        Blob: BlobProjector.ProjectedBlob  # Can't use the EntryPoint because of const-correctness.
    - Name: DescriptorAggregator
      Type: DescriptorAggregator
      Params: ~
      Inputs:
        Descriptors: >
          OrientedBoundingBoxSize.BoundingBoxSize
          OrientedNormalizedDensityHistogram.X
          OrientedNormalizedDensityHistogram.Y
          OrientedNormalizedDensityHistogram.Z
          HSVHistogram.Hue
          HSVHistogram.Saturation
          HSVHistogram.Value
          CloudOrienter.Eigenvalues
          CloudOrienter.RelativeCurvature
          
